spec/scenarios/charge_models/prorated_graduated_spec.rb

```diff
@@ -194,6 +194,112 @@
         end
       end
 
+      context 'when there are old events before first invoice' do
+        it 'returns expected invoice and usage amounts' do
+          Organization.update_all(webhook_url: nil) # rubocop:disable Rails/SkipsModelValidations
+          WebhookEndpoint.destroy_all
+
+          travel_to(DateTime.new(2023, 12, 1)) do
+            create_subscription(
+              {
+                external_customer_id: customer.external_id,
+                external_id: customer.external_id,
+                plan_code: plan.code,
+              },
+            )
+          end
+
+          subscription = customer.subscriptions.active.order(created_at: :desc).first
+
+          create(
+            :graduated_charge,
+            billable_metric:,
+            prorated: true,
+            plan:,
+            properties: {
+              graduated_ranges: [
+                {
+                  from_value: 0,
+                  to_value: 5,
+                  per_unit_amount: '0',
+                  flat_amount: '0',
+                },
+                {
+                  from_value: 6,
+                  to_value: nil,
+                  per_unit_amount: '12',
+                  flat_amount: '0',
+                },
+              ],
+            },
+          )
+
+          travel_to(DateTime.new(2023, 12, 2)) do
+            create_event(
+              {
+                code: billable_metric.code,
+                transaction_id: SecureRandom.uuid,
+                timestamp: 1699336493, ## November 2023
```

Comment:
Style/NumericLiterals: Use underscores(_) as thousands separator and separate every 3 digits with them. (https://rubystyle.guide#underscores-in-numerics)